### PR TITLE
Add Gyre manifest

### PR DESCRIPTION
**Pull request by HelpSeeker**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributible by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/HelpSeeker/Gyre
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

Not sure if it concerns the last point, but the manifest includes a custom FFmpeg build, as the one shipped with the GNOME platform runtime doesn't offer the functionality my program requires (most notably the concat demuxer). I made sure that it's the most minimal build configuration for my specific use case.

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: io.github.helpseeker.Gyre.json [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                    "url" : "https://github.com/HelpSeeker/Gyre.git",
+                    "tag" : "v0.1.0",
+                    "commit" : "a2419457cf98f7081012c37a492f37fb6fb7e2c9"
+                }
+            ]
+        }
+    ],
+    "build-options" : {
+        "env" : {        }
+    }
```
Comment (bilelmoussaoui) [inline io.github.helpseeker.Gyre.json:83]:
```suggestion
```

**Comment by bilelmoussaoui** [comment]
bot, build io.github.helpseeker.Gyre

**Comment by bilelmoussaoui** [comment]
bot, build io.github.helpseeker.Gyre

**Comment by TheEvilSkeleton** [comment]
@HelpSeeker you need a metadata file. You can check through [other flatpaks](https://github.com/flathub) and take a look at `metainfo.xml`.

**Comment by HelpSeeker** [comment]
Sorry, I'm a bit confused since I mostly just went with the structure created by GNOME Builder. I do have an io.github.helpseeker.Gyre.appdata.xml file, which gets installed under /app/share/appdata. So just changing the name of the directory from appdata to metainfo should do the trick, right? The file itself is now ok according to appstream-util.

**Comment by TheEvilSkeleton** [comment]
> So just changing the name of the directory from appdata to metainfo should do the trick, right?

Yes, it should do the trick. I'd suggest changing the name of the file to `io.github.helpseeker.Gyre.metainfo.xml` as well, to make it consistent to `/app/share/metainfo`.

By looking at your repo, you still don't have a metadata file, so keep that in mind. After that you can run `bot, build io.github.helpseeker.Gyre` and it'll work.

**Comment by barthalion** [comment]
You can use https://www.freedesktop.org/software/appstream/metainfocreator/#/ or https://bilelmoussaoui.github.io/metainfo-generator/ to generate it. It will likely fail to validate at first shot though, you can install `org.freedesktop.appstream-glib` from Flathub and use it from the terminal to validate the file locally.

**Comment by HelpSeeker** [comment]
Thank you both for your help. I have updated the metainfo file to pass the validation check and switched from the old appdata naming convention to metainfo. Hopefully this will fix all remaining issues.

**Comment by HelpSeeker** [comment]
bot, build io.github.helpseeker.Gyre

**Comment by TheEvilSkeleton** [comment]
It launched!

image

**Comment by nedrichards** [comment]
> Not sure if it concerns the last point, but the manifest includes a custom FFmpeg build, as the one shipped with the GNOME platform runtime doesn't offer the functionality my program requires (most notably the concat demuxer). I made sure that it's the most minimal build configuration for my specific use case.

No, that's fine! Thanks for the clarity though.

**Comment by barthalion** [comment]
I think the one thing I'd change for flatpak version is to ask for download directory on first launch – the default one seems to be inside the sandbox so all files are unreachable by players.

**Comment by HelpSeeker** [comment]
Thank you for the suggestion. I've planned to include an initial start screen for all versions.

**Pull request closed without merging**